tap_exacttarget/endpoints/campaigns.py

```diff
@@ -44,7 +45,9 @@ def sync_data(self):
             FuelSDK.ET_Campaign,
             self.auth_stub)
 
+        catalog_copy = copy.deepcopy(self.catalog)
```

Comment:
Why do a deep copy of the catalog?

Reply:
The behavior of the transform function is in such a manner ([here](https://github.com/singer-io/singer-python/blob/0c066de21111d8572425083b4a8792d193c80af1/singer/transform.py#L169)) that when we pass catalog in transform function, it is re-ordering the data-type values for eg. before transform `["null", "string"]` and after transform `["string", "null"]`. 
So we made a copy of the catalog to save the original catalog as we are using `self.catalog` for writing the schema.